queries/rust/textobjects.scm

```diff
@@ -47,26 +47,13 @@
 
 (match_expression) @conditional.outer
 
-(if_let_expression
-  consequence: (block)?
-  @conditional.inner) @conditional.outer
-
-(if_let_expression
-  alternative: (else_clause (block) @conditional.inner))
-
-(if_let_expression
-  pattern: (_) @conditional.inner)
```

Comment:
I think, that this node isn't handled in the normal `if_expression` path.

Reply:
The equivalent to this should be:

```
(if_expression
  condition: (let_condition pattern: (_) @conditional.inner))
```

But I'm not sure if this would clash with this one:

```
(if_expression
  condition: (_) @conditional.inner)
```

If I understand this correctly we would now assign `conditional.inner` to two different parts of the AST. So this might not work. But again I'm not that deep into tree-sitter semantics.

Reply:
With this PR and considering the following code.

```rust
if let Some(i) = emoticon {
    println!("Matched {:?}!", i);
}
```

`@conditional.inner` will capture `let Some(i) = emoticon` and `@conditional.outer` will select the whole if expression, which makes sense to me. What do you think?

Reply:
> But I'm not sure if this would clash with this one:

It doesn't - if the cursor is inside `Some(foo)` in `if let Some(foo) = None {}` it matches only the pattern, if your cursor is outside it but inside `let Some(foo) = None` it matches the whole condition